server/apps/search/__init__.py

```diff
@@ -8,34 +8,41 @@
 # AUTHORS and LICENSE files distributed with this source code, or
 # at https://www.sourcefabric.org/superdesk/license
 
+from flask import current_app as app, json, g
+from eve_elastic.elastic import set_filters
 
 import superdesk
-from flask import current_app as app, json, g
 from apps.archive.common import aggregations
-from eve_elastic.elastic import set_filters
+from superdesk.metadata.item import CONTENT_STATE, ITEM_STATE
+from apps.archive.archive import SOURCE as ARCHIVE
 
 
 class SearchService(superdesk.Service):
-    """Federated search service.
-
-    It can search ingest/content/archive/spike at the same time.
     """
+    Federated search service.
 
-    available_repos = ('ingest', 'archive', 'text_archive', 'published')
-    default_repos = ['ingest', 'archive', 'text_archive', 'published']
+    It can search against different collections like Ingest, Production, Archived etc.. at the same time.
+    """
 
-    private_filters = [{
-        'or': [
-            {'and': [{'exists': {'field': 'task.desk'}},
-                     {'terms': {'state': ['fetched', 'routed', 'draft', 'in_progress', 'submitted']}}]},
-            {'not': {'term': {'_type': 'archive'}}},
-            {'and': [{'term': {'_type': 'published'}},
-                     {'terms': {'state': ['published', 'killed']}}]}
-        ]
-    }]
+    def __init__(self, datasource, backend):
+        super().__init__(datasource=datasource, backend=backend)
+        self._default_repos = ['ingest', 'archive', 'published', 'archived']
 
-    def get_private_filter(self):
-        pass
+        self._private_filters = [{
+            'or': [
+                {'and': [{'term': {'_type': 'ingest'}}]},
+                {'and': [{'exists': {'field': 'task.desk'}},
+                         {'terms': {ITEM_STATE: [CONTENT_STATE.FETCHED, CONTENT_STATE.ROUTED, CONTENT_STATE.DRAFT,
```

Comment:
I think the search should return the stories from that user's workspace as well. This was how it was before I guess...